### DISCUSSION TITLE
Multiple overlapping results from EntityRuler

### DISCUSSION THREAD

**Discussion by rubmz** [explosion/spaCy#6050] [New Features & Project Ideas]
## Feature description
When doing rule-based entity matching there's a need to score overlapping results and compare the best option based on various criteria. Problem is: EntityRuler overrides existing entities and won't give all the possible patterns that may match some span.
An even better option would be to supply a scoring function along with the patterns, taking the highest score as the winning match...

Guess this can be implemented as an external [custom component](https://spacy.io/usage/processing-pipelines#custom-components) or [spaCy plugin](https://spacy.io/universe).

**Comment by rubmz** [comment]
After looking into the implementation details of EntityRuler, seems better just use a Matcher with callback. But this feature still is required IMHO.

**Comment by adrianeboyd** [comment]
Yes, the `EntityRuler` is basically just a wrapper around the `Matcher` and/the `PhraseMatcher`, so you can use those directly with your own filtering. There are so many different scoring options that I think it's going to be tricky to add a generic and useful scoring option to the `EntityRuler` (handling the serialization is the tricky part), but we can keep this in mind as a potential feature.
